src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Runtime/PreferStreamReadAsyncMemoryOverloadsTests.cs

```diff
@@ -780,6 +780,47 @@ public async void M()
             return CSharpVerifyExpectedCodeFixDiagnosticsAsync(originalSource, fixedSource, GetCSharpResult(12, 74, 12, 254));
         }
 
+
+        [Fact]
+        public Task CS_Fixer_PreserveNullability()
+        {
+            string originalSource = @"
+#nullable enable
+using System;
+using System.IO;
+using System.Threading.Tasks;
+
+public class C
+{
+    async void M(FileStream? stream, byte[]? buffer)
+    {
+        await stream!.ReadAsync(buffer!, offset: 0, count: buffer!.Length).ConfigureAwait(false);
```

Comment:
Do we have some test-cases with something like `buffer?.Length ?? 0`

Reply:
I added them. Can you please take another look?